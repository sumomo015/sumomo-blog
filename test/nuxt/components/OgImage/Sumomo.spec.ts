// @vitest-environment nuxt

import { describe, it, expect } from 'vitest'
import { mountSuspended } from '@nuxt/test-utils/runtime'

import OgImageSumomo from '~/components/OgImage/Sumomo.vue'

describe('OgImageSumomo', () => {
  const selectorTitle = '[data-testid="title"]'
  const selectorDescription = '[data-testid="description"]'
  const selectorAuthor = '[data-testid="author"]'

  it('タイトルが表示されること', async () => {
    const wrapper = await mountSuspended(OgImageSumomo, {
      props: { title: 'Test Title' },
    })
    expect(wrapper.find(selectorTitle).text()).toBe('Test Title')
  })

  it('説明が表示されること', async () => {
    const wrapper = await mountSuspended(OgImageSumomo, {
      props: { description: 'Test Description' },
    })
    expect(wrapper.find(selectorDescription).text()).toBe('Test Description')
  })

  it('説明が渡されない場合は表示されないこと', async () => {
    const wrapper = await mountSuspended(OgImageSumomo)
    expect(wrapper.find(selectorDescription).exists()).toBe(false)
  })

  it('署名が表示されること', async () => {
    const wrapper = await mountSuspended(OgImageSumomo)
    expect(wrapper.find(selectorAuthor).text()).toBe('スモモ')
  })

  it('スナップショット', async () => {
    const wrapper = await mountSuspended(OgImageSumomo, {
      props: {
        title: 'Test Title',
        description: 'Test Description',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
