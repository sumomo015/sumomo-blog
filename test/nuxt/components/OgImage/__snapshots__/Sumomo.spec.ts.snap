// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OgImageSumomo > スナップショット 1`] = `
"<div class="relative h-full w-full flex items-center justify-center border-2 border-white bg-neutral-900 text-white">
  <div class="flex flex-col items-center gap-4 text-center">
    <h1 data-testid="title" class="flex text-7xl font-bold">Test Title</h1>
    <p data-testid="description" class="max-w-3xl text-2xl">Test Description</p>
  </div>
  <p data-testid="author" class="absolute bottom-17 right-13 flex gap-4 font-bold"> スモモ </p>
</div>"
`;
